crmsh/ui_sbd.py

```diff
@@ -0,0 +1,528 @@
+        '''
+        update_dict = {}
+        parameter_dict = self._adjust_timeout_dict(parameter_dict, diskless=True)
+        watchdog_timeout = parameter_dict.get("watchdog")
+        if watchdog_timeout and watchdog_timeout != self.watchdog_timeout_from_config:
+            update_dict["SBD_WATCHDOG_TIMEOUT"] = str(watchdog_timeout)
+        watchdog_device = parameter_dict.get("watchdog-device")
+        if watchdog_device != self.watchdog_device_from_config:
+            update_dict["SBD_WATCHDOG_DEV"] = watchdog_device
+
+        sbd_manager = sbd.SBDManager(
+            update_dict=update_dict,
+            diskless_sbd=True
+        )
+        sbd_manager.init_and_deploy_sbd()
+
+    def _device_add(self, devices_to_add: typing.List[str]):
+        '''
+        Implement sbd device add command, add devices to sbd configuration
+        '''
+        all_device_list = self.device_list_from_config + devices_to_add
+        sbd.SBDUtils.verify_sbd_device(all_device_list)
+
+        devices_to_init, _ = sbd.SBDUtils.handle_input_sbd_devices(
+            devices_to_add,
+            self.device_list_from_config
+        )
+
+        logger.info("Append devices: %s", ';'.join(devices_to_add))
+        update_dict = {"SBD_DEVICE": ";".join(all_device_list)}
+        sbd_manager = sbd.SBDManager(
+            device_list_to_init=devices_to_init,
+            update_dict=update_dict,
+            timeout_dict=self.device_meta_dict_runtime
+        )
+        sbd_manager.init_and_deploy_sbd()
+
+    def _device_remove(self, devices_to_remove: typing.List[str]):
+        '''
+        Implement sbd device remove command, remove devices from sbd configuration
+        '''
+        for dev in devices_to_remove:
+            if dev not in self.device_list_from_config:
+                raise self.SyntaxError(f"Device {dev} is not in config")
+        # To keep the order of devices during removal
+        left_device_list = [dev for dev in self.device_list_from_config if dev not in devices_to_remove]
+        if len(left_device_list) == 0:
+            raise self.SyntaxError("Not allowed to remove all devices")
+
+        logger.info("Remove devices: %s", ';'.join(devices_to_remove))
+        update_dict = {"SBD_DEVICE": ";".join(left_device_list)}
+        sbd.SBDManager.update_sbd_configuration(update_dict)
+        sbd.SBDManager.restart_cluster_if_possible()
+
+    @command.completers_repeating(sbd_device_completer)
+    def do_device(self, context, *args) -> bool:
+        '''
+        Implement sbd device command
+        '''
+        if not ServiceManager().service_is_active(constants.PCMK_SERVICE):
+            logger.error("%s is not active", constants.PCMK_SERVICE)
+            return False
+        if not sbd.SBDUtils.is_using_disk_based_sbd():
+            logger.error("Only works for disk-based SBD")
+            logger.info("Please use 'crm cluster init sbd' to configure disk-based SBD first")
```

Comment:
Changed as:
```
INFO: Please use 'crm cluster init -s <dev1> [-s <dev2> [-s <dev3>]]' to configure disk-based SBD first
```
Since the given device path might be very long